File: benchmark-java-sync/src/main/java/com/example/benchmarkjavasync/core/ApplicationRepository.java
package com.example.benchmarkjavasync.core;

import com.example.benchmarkjavasync.core.model.CreditInformation;
import com.example.benchmarkjavasync.core.model.Passport;
import com.example.benchmarkjavasync.core.model.User;

public interface ApplicationRepository<R, V> {

    R saveUser(User user);

    V saveCreditInformation(CreditInformation creditInformation);

    V savePassport(Passport passport);
}

File: benchmark-kotlin-async/src/main/kotlin/com/example/benchmarkkotlinasync/core/ApplicationRepository.kt
package com.example.benchmarkkotlinasync.core;

import com.example.benchmarkkotlinasync.core.model.CreditInformation
import com.example.benchmarkkotlinasync.core.model.Passport
import com.example.benchmarkkotlinasync.core.model.User

interface ApplicationRepository<R, V> {

    suspend fun saveUser(user: User): R

    suspend fun saveCreditInformation(creditInformation: CreditInformation): V

    suspend fun savePassport(passport: Passport): V
}
